scripts/run_e2e_tests.py

```diff
@@ -446,18 +446,41 @@ def get_chrome_driver_version():
     This method follows the steps mentioned here:
     https://chromedriver.chromium.org/downloads/version-selection
     """
-    try:
-        proc = subprocess.Popen(
-            ['google-chrome', '--version'], stdout=subprocess.PIPE)
-        output = proc.stdout.readline()
-    except OSError:
-        raise Exception(
-            'Failed to execute "google-chrome --version" command. This is '
-            'used to determine the chromedriver version to use. Please set '
-            'the chromedriver version manually using --chrome_driver_version '
-            'flag. To determine the chromedriver version to be used, please '
-            'follow the instructions mentioned in the following URL:\n'
-            'https://chromedriver.chromium.org/downloads/version-selection')
+    if common.is_mac_os():
+        try:
```

Comment:
You should be able to reduce the branching significantly by having "if common.is_mac_os()" apply only to the string/array that is passed to Popen (and that is used in the error message). 

So, declare a local variable for that string (chrome_command = ... if common.is_mac_os() else ...) and then you don't need the big if/else. In general, reduce branching where possible since that adds contextual overhead.

Reply:
Thanks Sean! I figured out that when I removed the backslashes for the popen command it works better and you don't need shell=True so that made it way cleaner :) Thanks so much :D